daemon/proto/40swapd.proto

```diff
@@ -2,43 +2,98 @@ syntax = "proto3";
+  rpc GetSwapOut(GetSwapOutRequest) returns (GetSwapOutResponse); // Retrieves the status of a SwapOut.
 }
 
-// Enum definition
+// Enum definition for supported blockchain chains.
 enum Chain {
-  BITCOIN = 0;
-  LIQUID = 1;
+  BITCOIN = 0; // Bitcoin blockchain.
+  LIQUID = 1;  // Liquid sidechain.
 }
 
+// Enum definition for supported network types.
 enum Network {
-  MAINNET = 0;
-  TESTNET = 1;
-  REGTEST = 2;
+  MAINNET = 0; // Main Bitcoin network.
+  TESTNET = 1; // Bitcoin test network.
+  REGTEST = 2; // Bitcoin regression test network.
+}
+
+// Enum definition for swap statuses.
+enum Status {
+  // Happy path statuses.
+  CREATED = 0; // Swap has been created.
+  INVOICE_PAYMENT_INTENT_RECEIVED = 1; // Payment locked on L2
+  CONTRACT_FUNDED_UNCONFIRMED = 2; // Contract funded but unconfirmed (mempool)
+  CONTRACT_FUNDED = 3; // Contract funded and confirmed.
+  INVOICE_PAID = 4; // L2 invoice has been paid.
+  CONTRACT_CLAIMED_UNCONFIRMED = 5; // Contract claimed but unconfirmed.
+  DONE = 6; // Swap completed successfully.
+
+  // Expiry-related statuses.
+  CONTRACT_REFUNDED_UNCONFIRMED = 7; // Contract refunded but unconfirmed.
+  CONTRACT_EXPIRED = 8; // Contract expired.
 }
 
-// Message definitions
+// Message definitions for SwapIn operation.
 message SwapInRequest {
-  Chain chain = 1;
-  optional string invoice = 2;
-  optional uint32 amountSats = 3;
-  optional uint32 expiry = 4;
+  Chain chain = 1; // Blockchain chain for the swap.
+  optional string invoice = 2; // Invoice to be paid.
+  optional uint32 amount_sats = 3; // Amount in satoshis.
+  optional uint32 expiry = 4; // Expiry time for the swap.
 }
 
 message SwapInResponse {
-  string swapId = 1;
-  string claim_address = 2;
+  string swap_id = 1; // Unique identifier for the swap.
+  string claim_address = 2; // Address to claim the funds.
 }
 
+// Message definitions for SwapOut operation.
 message SwapOutRequest {
-  Chain chain = 1;
-  // Amount in satoshis
-  uint64 amountSats = 2;
-  // Optional Destination address
-  string address = 3;
+  Chain chain = 1; // Blockchain chain for the swap.
+  uint64 amount_sats = 2; // Amount in satoshis.
+  string address = 3; // Optional destination address.
+}
+
+message SwapOutResponse {
+
 }
 
-message SwapOutResponse {}
\ No newline at end of file
+// Message definitions for querying SwapIn status.
+message GetSwapInRequest {
+  string id = 1; // Unique identifier for the swap.
+}
+
+message GetSwapInResponse {
+  string id = 1; // Unique identifier for the swap.
+  string contract_address = 2; // Address of the contract.
+  google.protobuf.Timestamp created_at = 3; // Timestamp when the swap was created.
+  double input_amount = 4; // Input amount in BTC.
+  optional string lock_tx = 5; // Lock transaction txid.
+  optional string outcome = 6; // Outcome of the swap.
+  double output_amount = 7; // Output amount in BTC.
```

Comment:
should we change the other requests to btc to have a standard amount type?

Reply:
@RodriFS that's a good question but I would say for integrations sats it's easier, we use BTC because the api GET endpoints use BTC

I will move along and create issues

Reply:
#63